### ISSUE TITLE
Improve support for a mish-mash video folder structure

### ISSUE THREAD

**Issue by LukePulverenti** [MediaBrowser/Emby#3207]
That doesn't follow any naming guidelines.

**Comment by anoma** [comment]
Is this related to https://github.com/MediaBrowser/Emby/issues/3205 and if so should I close that one with this superseding it?

**Comment by LukePulverenti** [comment]
@anoma I haven't looked at your examples in depth yet, but if your folder structure does not match what is listed in our tv naming guide, then yes it is related.

**Comment by CatSamaaa** [comment]
I have a “Books” library.
And I have some audio books there.
Now I found that if I put some audios and a pdf (the scenario of the audio book) in a folder, Emby scaner will ignore all the audios.



you can reappear this BUG by create a floder with this folder structure
![image](https://user-images.githubusercontent.com/34727840/38789452-d8731412-416c-11e8-915d-1243d426dd0c.png)

**Comment by CorDawg** [comment]
Hello, I see this has been a pain in many people sides for months, do you have a possible eta for when this feature will be available or should I just give up on emby and try something else (Plex mobile streaming is super bad). My use case has TV shows being purged after 30 days so setting up seperate folders is completely pointless, I have no intentions of creating seperate folders to keep these files that I will never watch again.

Anyway tldr is there a possible ETA on implementing Mish Mash folder.

Thanks,
CJ

**Comment by nomandera** [comment]
I dont know if something has changed recently but as off Version 4.1.1.0 the TV show view is truely terrible. It looks worse than it ever did.

To use similar wording as https://github.com/jellyfin/jellyfin/issues/1035  `folders that contain inner tv folders are themselves considered as seperate tv shows`.

The net result of this is Emby looks appalling compared to say Kodi. My collection of ripped TV shows often has 20...40...60 plus bad show matches listed instead of one.

Can I suggest as a interim fix that isnt so "big" that tvshow.nfo and movie.nfo be used as anchors signifying that everything under this folder is the same show or movie. This also has the benefit of being useful in itself as a means to manually fix or tag content 100% correctly identified for all time (and is in fact what other projects have already implemented)

Update: I just noticed that if I manually fix any of the episodes so match Emby standard naming/folder convention then even the ones I dont fix are identified correctly e.g. fix one the other 20 are tagged even without applying the naming fix. Obviously something has changed at some point, caused these mismatches but the later fix hasnt picked up all this bad matched content.

Update 2: **I have found the bug and a work around fix** (and the reason I could not replicate this easily). I am unsure if this needs a new ticket but will just stay here for now.

If a TV show contains 100% subfolders that are episodic then Emby completely fails to properly match them

e.g.

- /show/episode.s01e01/episode.s01e01.mkv
- /show/episode.s01e02/episode.s01e02.mkv

will be incorrectly matched as two separate unknown shows. Even if you delete the show and re-add it or refresh it in any other way the situation persists.


However if you subsequently fix or add ANY single episode so that resides in a season folder then that episode AND ALL untouched/unfixed episodes are then matched right away by Emby.

e.g.

- /show/episode.s01/episode.s01e01.mkv
- /show/episode.s01e02/episode.s01e02.mkv

will be correctly matched as one show with separate episodes even though one file remains identical to the initial case where Emby could not match it.

**Comment by LukePulverenti** [comment]
There is a regression in 4.1 related to nested episode folders, and this is resolved in the 4.2 beta channel. Thanks.

**Comment by nomandera** [comment]
Thanks for taking the time to post. Given that I expended quite a bit of effort running scenario after scenario to debug this issue and make a report, which is now wasted given its a known bug with an existing fix, where should I have been tracking to avoid this?

Apologies if this is obvious but Emby has many forums and many repos and it is not at all obvious where I should be following. Even now I cant find a ticket, PR or code comment addressing this bug.

**Comment by LukePulverenti** [comment]
This is coming to Emby Server 4.6 for music libraries. Everything will be fully tag-driven and any folder structure will now be accepted. A test build will be available in the next day or two:
https://emby.media/community/index.php?/topic/92417-4607-music-scanning-rewrite/
Once we get through this, then we'll apply the same concepts and add the same kind of support for TV libraries.

**Comment by EnigmaSolved** [comment]
> Once we get through this, then we'll apply the same concepts and add the same kind of support for TV libraries.

Interesting. I'm curious how this would work for video files? Like, will we be tagging the files (similar to how audio file tagging works)?

**Comment by LukePulverenti** [comment]
No, you won't be. Video libraries will add some additional challenges on top of this, so it's not quite a 1:1 fit, but some of what we're doing with music can be transferred over.

**Comment by EnigmaSolved** [comment]
> 
> 
> No, you won't be. Video libraries will add some additional challenges on top of this, so it's not quite a 1:1 fit, but some of what we're doing with music can be transferred over.

Okay, that makes sense. I was under the impression that video tag functionality in operating systems wasn't terribly robust (in contrast to music tagging).

**Comment by LukePulverenti** [comment]
Note to self of an example to test when this gets done for TV: https://emby.media/community/index.php?/topic/93262-tv-shows-with-folder-names-and-jibberish-mkv-file-names/&do=findComment&comment=963352
